backend/chunker/chunker.go

```diff
@@ -1077,7 +1106,7 @@ func (f *Fs) put(ctx context.Context, in io.Reader, src fs.ObjectInfo, remote st
 	switch f.opt.MetaFormat {
 	case "simplejson":
 		c.updateHashes()
-		metadata, err = marshalSimpleJSON(ctx, sizeTotal, len(c.chunks), c.md5, c.sha1)
+		metadata, err = marshalSimpleJSON(ctx, sizeTotal, len(c.chunks), c.md5, c.sha1, xactID)
```

Comment:
`marshalJson` is the commit point of this operation when external actors and parallel operations start seeing changes (here it's the uploaded chunks).
With meta v1 we should have removed old permanent chunks and renamed new temporary chunks by far.
With meta v2 (when transaction id is non-empty string):
- there is no need to rename new chunks (hurray)
- old chunks become garbage automatically when meta is updated, so we can postpone their removal until return and let others see changes earlier

The above is not true if `useMeta==false` or for non-composite objects (aka wrappers).

Reply:
More precisely, the very commit point is here: https://github.com/rclone/rclone/blob/v1.53.2/backend/chunker/chunker.go#L1084

Reply:
> we can postpone their removal until return and let others see changes earlier

Correcting myself.
For this optimization the `removeOldChunks()` will have to know the old transaction id (pre commit point).

Reply:
@mcalman @leoluan1 Feel free to skip the `removeOldChunks` optimization, it's optional for this PR.